app.js

```diff
@@ -0,0 +1,141 @@
+import { app, errorHandler } from 'mu';
+import VRDocumentName from './lib/vr-name';
+import CONSTANTS from './constants';
+import { getOrderedAgendaitems } from './lib/ordered-agendaitems';
+import { getSubcasePieces } from './lib/subcase-pieces';
+import { addAgendaSubcaseId } from './lib/add-agenda-subcase-id';
+import { addPiecePosition } from './lib/add-piece-position';
+import { addPieceOriginalName } from './lib/add-piece-original-name';
+
+app.get('/', async function (req, res) {
+  res.sendStatus(202);
+
+  const results = await getOrderedAgendaitems(2024);
+
+  const knownSubcases = new Set();
+  const errors = [];
+  const addError = (thing) => errors.push(thing);
+
+  const counters = {
+    regular: {
+      doc: 0,
+      med: 0,
+      dec: 0,
+    },
+    pvv: {
+      doc: 0,
+      med: 0,
+      dec: 0,
+    },
+  }
+  for (const result of results) {
+    const {
+      subcase,
+      agendaitemType,
+      subcaseType,
+      meetingType,
+    } = result;
+
+    if (knownSubcases.has(subcase)) {
+      console.debug(`Skipping subcase <${subcase}>`);
+      continue;
+    }
+    knownSubcases.add(subcase);
+
+    console.debug(result);
+
+    const piecesResults = await getSubcasePieces(subcase);
+    let hasDecreetPiece = false;
+    for (const { pieceType } of piecesResults) {
+      if (pieceType === CONSTANTS.PIECE_TYPES.DECREET) {
+        hasDecreetPiece = true;
+      }
+    }
+
+    const isDoc = agendaitemType === CONSTANTS.AGENDA_ITEM_TYPES.NOTA;
+    const isMed = agendaitemType === CONSTANTS.AGENDA_ITEM_TYPES.MEDEDELING;
+    const isDec = agendaitemType === CONSTANTS.AGENDA_ITEM_TYPES.NOTA && subcaseType === CONSTANTS.SUBCASE_TYPES.BEKRACHTIGING && hasDecreetPiece;
+
+    console.debug('isDoc:', isDoc);
+    console.debug('isMed:', isMed);
+    console.debug('isDec:', isDec);
+
+    // Counters
+    let counter;
+    if (meetingType === CONSTANTS.MEETING_TYPES.PVV) {
+      counter = 1 + (isDec ? counters.pvv.dec : isDoc ? counters.pvv.doc : counters.pvv.med);
+    } else {
+      counter = 1 + (isDec ? counters.regular.dec : isDoc ? counters.regular.doc : counters.regular.med);
+    }
+    const pieces = [];
+    let counterMatched = false;
+    for (const { piece, pieceName } of piecesResults) {
+      let vrDoc = null;
+      try {
+        vrDoc = new VRDocumentName(pieceName);
+      } catch {
+        console.debug(`Could not parse piece name ${pieceName}. Probably a piece from testing, skipping`);
+        continue;
+      }
+      pieces.push({ piece, pieceName, vrDoc });
+
+      // Validation
+      if (isDoc && isMed) {
+        addError({ subcase, counter, vrDoc, message: 'Marked as a NOTA and a MEDEDLING' });
+      }
+
+      if (isDec && isMed) {
+        addError({ subcase, counter, vrDoc, message: 'Marked as a DECREET and a MEDEDELING' });
+      }
```

Comment:
Can't this be taken out of this loop? Also don't see how `isDoc` and `isMed` can be true at the same time